lib/env_irb_prompt.rb

```diff
@@ -0,0 +1,36 @@
+require 'socket'
+
+class EnvIrbPrompt
+  # @param [Hash] irb_conf
+  def configure!(irb_conf = IRB.conf)
+    irb_conf[:USE_AUTOCOMPLETE] = false
+    irb_conf[:SAVE_HISTORY] = on_deployed_box? ? nil : 1000
+
+    irb_conf[:PROMPT][:ENV_PROMPT] = {
+      PROMPT_I: "%N(#{bold(env)}):%03n:%i> ",
+      PROMPT_S: "%N(\e[1m#{bold(env)}\e[22m):%03n:%i%l ",
+      PROMPT_C: "%N(\e[1m#{bold(env)}\e[22m):%03n:%i* ",
+      RETURN: "%s\n",
+    }
+    irb_conf[:PROMPT_MODE] = :ENV_PROMPT
+  end
+
+  def on_deployed_box?
+    return @on_deployed_box if defined?(@on_deployed_box)
+    @on_deployed_box = File.directory?('/srv/idp/releases/')
```

Comment:
On some rollplans, I've seen `cd /srv/idp/current`. Is that a symbolic link to `/srv/idp/releases`? Is this a reliable way to tell that we're on a deployed box?

Reply:
`/current` is a symlink to a folder inside `releases` and was reliable enough for our current check, so I opted to keep it:

https://github.com/18F/identity-idp/blob/15f9b25d7106da90020c80819cd0c316086b2844/.irbrc#L3

previously: https://github.com/18F/identity-idp/pull/8147